src/render/filters.rs

```diff
@@ -234,6 +235,70 @@ impl ResolveFilter for ExternalFilter {
     }
 }
 
+impl ResolveFilter for FirstFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        _template: TemplateString<'t>,
+        _context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let content = match variable {
+            Some(content) => {
+                // Try to get the first item from the content
+                match content {
+                    Content::Py(obj) => {
+                        // Try to get the first item using Python's indexing
+                        // Django only catches IndexError, not TypeError
+                        match obj.get_item(0) {
+                            Ok(first) => Some(Content::Py(first)),
+                            Err(e) => {
+                                // Check if it's an IndexError
+                                if e.is_instance_of::<pyo3::exceptions::PyIndexError>(py) {
+                                    // IndexError -> return empty string
+                                    "".as_content()
+                                } else {
+                                    // Other errors (like TypeError) -> propagate
+                                    return Err(e.into());
+                                }
+                            }
+                        }
+                    }
+                    Content::String(s) => {
+                        // For strings, get the first character
+                        let s_raw = s.as_raw();
+                        if s_raw.is_empty() {
+                            "".as_content()
+                        } else {
+                            // Get the first character
+                            let first_char = s_raw.chars().next().unwrap();
+                            first_char.to_string().into_content()
+                        }
```

Comment:
I think we can be more efficient here with the [`str.split` method](https://doc.rust-lang.org/std/primitive.str.html#method.split), which returns an iterator of `&str` slices, avoiding allocating a `char` or a `String`:

```suggestion
                        // Skip the empty string always present at the start after splitting
                        let chars = s.split("").skip(1);
                        match chars.next() {
                            None => "".as_content(),
                            c => c.as_content(),
                        }
```